pilot/pkg/model/push_context.go

```diff
@@ -502,11 +502,21 @@ func (ps *PushContext) DestinationRule(proxy *Proxy, service *Service) *Config {
 		return nil
 	}
 
-	// search through the DestinationRules in proxy's namespace first
-	if ps.namespaceLocalDestRules[proxy.ConfigNamespace] != nil {
-		if host, ok := MostSpecificHostMatch(service.Hostname,
-			ps.namespaceLocalDestRules[proxy.ConfigNamespace].hosts); ok {
-			return ps.namespaceLocalDestRules[proxy.ConfigNamespace].destRule[host].config
+	// TODO: once we move default DestinationRule to root config namespace,
+	// remove the hacky check below.  Until then, we need an exception for
+	// istio-system namespace, as it happens to be the home of global dest
+	// rules for mTLS, as well as the home of the gateway proxy. The root
+	// config namespace is technically not supposed to have any proxy
+	// istio-ingressgateway and istio-egressgateway wont hit this if block.
+	// instead, dest rules will first be looked up from the service's own namespace
+	// if not found, we look at ALL dest rules across all namespaces. This is the old behavior
+	if proxy.ConfigNamespace != IstioSystemNamespace {
```

Comment:
ConfigNamespace should be istio-system, never istio-system. When would this happen ?